Sources/Common/Store/SdkConfig.swift

```diff
@@ -14,11 +14,47 @@ import Foundation
  ```
  */
 public struct SdkConfig {
+    public enum Environment {
+        static let siteId = "siteId"
+        static let apiKey = "apiKey"
+        static let region = "region"
+    }
+
+    public enum Config {
+        static let trackingApiUrl = "trackingApiUrl"
+        static let autoTrackDeviceAttributes = "autoTrackDeviceAttributes"
+        static let logLevel = "logLevel"
+        static let autoTrackPushEvents = "autoTrackPushEvents"
+        static let backgroundQueueMinNumberOfTasks = "backgroundQueueMinNumberOfTasks"
+        static let backgroundQueueSecondsDelay = "backgroundQueueSecondsDelay"
+    }
+
     // Used to create new instance of SdkConfig when the SDK is initialized.
     // Then, each property of the SdkConfig object can be modified by the user.
     public enum Factory {
-        public static func create(region: Region) -> SdkConfig {
-            SdkConfig(trackingApiUrl: region.productionTrackingUrl)
+        public static func create(region: Region, params: [String: Any] = [:]) -> SdkConfig {
+            var config = SdkConfig(trackingApiUrl: region.productionTrackingUrl)
```

Comment:
Wondering if `_sdkWrapperConfig` has to be handled form wrappers or it's value also to be set within the native modules. I think the latter one can be done by feeding the values in the parameter itself. Has that been considered or mistakenly missed out ?

Reply:
I kept it in wrappers because I feel like, that logic depends very much on wrappers, and it's not shared, e.g, React also caters to the case of expo while flutter doesn't care about it. 

For now, I just pushed the common actors so no logic is done by Native, but open to adding it if helpful.

Reply:
IMO, we should add `_sdkWrapperConfig` to the common module. Distributing configuring the SDK based on just one property doesn't sound good to me. 

You are right, that in react native wrapper we are handling expo and react native both, but we pass only one value to `SdkWrapperConfig` based on condition

```
 var sdkSource = SdkWrapperConfig.Source.reactNative
        if source.lowercased() == "expo" {
            sdkSource = SdkWrapperConfig.Source.expo
        }
```
`SdkWrapperConfig(source: sdkSource, version: pversion )`

Reply:
Added